juju/juju
# state: stop using ForModel in Model methods

## Description of change

Model methods were using ForModel if the
State that returned the Model was not opened
for the model UUID. This is expensive.

Instead, use the Database and copy for the
model UUID as necessary. Many of the uses
do not even require this, as they operate
on global collections only. Making this change
requires that various functions (e.g. reading
settings) operate on a Database, and not a
*State.

## QA steps

1. juju bootstrap, deploy ubuntu (smoke test)
2. run all the unit tests
3. run "juju-introspect" on the controller, observe no "started state for model..." messages in the log

## Documentation changes

None.

## Bug reference

Fixes https://bugs.launchpad.net/juju/+bug/1570269.
